schemes/musica/tuvx/musica_ccpp_tuvx.F90

```diff
@@ -62,31 +64,57 @@ subroutine tuvx_init(vertical_layer_dimension, vertical_interface_dimension, &
       return
     end if
 
-    radiators => radiator_map_t( error )
+    temperature_profile => create_temperature_profile( height_grid, errmsg, errcode )
+    if (errcode /= 0) then
+      deallocate( grids )
+      deallocate( height_grid )
+      height_grid => null()
+      deallocate( profiles )
```

Comment:
There seems to be a lot of code duplication in the `deallocate` / `=> null()` clean-up calls here whenever an error occurs. Would it be possible to wrap this into a clean-up subroutine that deallocates/nullifies the pointers in this module if allocated?

Reply:
Yes, we wanted to address that and I've opened a small PR to specifically for that. I'll add you as a reviewer. Would you be able to review it?

Reply:
Done, thank you!